### PR TITLE
Fixing import getting stuck on step 4 

### LINKED ISSUE #12624: Flat file import stuck on Step 4
- Azure Data Studio Version:

Steps to Reproduce:

1. at the step 4, import status takes too long to load, been 30 minutes and still stuck. This happens right after the update
test

### LINKED ISSUE #12648: Wont let me import a flat file when localhost is connected
Issue Type: <b>Bug</b>

It's frustrating that i cant import my .csv file, it stays on import status and wont go through, I'vve done it before and worked perfectly, don't know whats going this time. 

Azure Data Studio version: azuredatastudio 1.22.0 (77b9a708df3679c26a89ba24a070b3f7a078a510, 2020-09-21T18:41:27.116Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz (8 x 2500)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.09GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
import|Mic|1.0.0

</details>

### PR DESCRIPTION
**Pull request by aasimkhan30**
Exposing errors of change column settings and stopping import if they occur

This PR fixes #12624, fixes #12648.

### CODE REVIEW

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
You have a test failure now

File: extensions/import/src/wizard/pages/summaryPage.ts [hunk 1]
```diff
@@ -117,7 +117,19 @@ export class SummaryPage extends ImportPage {
...
+		if (changeColumnSettingsErrors.length !== 0) {
+			let updateText: string;
+			updateText = changeColumnSettingsErrors.join(EOL);
+			this.statusText.updateProperties({
+				value: updateText
+			});
+			return true;
 		}
 
 		let result: InsertDataResponse;
```
[the remark this answers is not in the document]
Comment (aasimkhan30) [reply extensions/import/src/wizard/pages/summaryPage.ts:135]: There seems to be a mistake. The correct property is authenticationType

File: extensions/import/src/wizard/pages/summaryPage.ts [hunk 2]
```diff
@@ -117,7 +117,19 @@ export class SummaryPage extends ImportPage {
...
 			};
 			const changeColumnResult = await this.provider.sendChangeColumnSettingsRequest(columnChangeParams);
-			changeColumnResults.push(changeColumnResult);
+			if (changeColumnResult.result.errorMessage) {
+				changeColumnSettingsErrors.push(changeColumnResult.result.errorMessage);
+			}
+		}
+
+		// Stopping import if there are errors in change column setting.
+		if (changeColumnSettingsErrors.length !== 0) {
```
[the remark this answers is not in the document]
Comment (aasimkhan30) [reply extensions/import/src/wizard/pages/summaryPage.ts:126]: The service in this call only sets up some local variables. So, there isn't any good actionable error message to show to the users.

File: extensions/import/src/wizard/pages/summaryPage.ts [hunk 3]
```diff
@@ -117,7 +117,19 @@ export class SummaryPage extends ImportPage {
 				newInPrimaryKey: val.primaryKey
 			};
 			const changeColumnResult = await this.provider.sendChangeColumnSettingsRequest(columnChangeParams);
-			changeColumnResults.push(changeColumnResult);
+			if (changeColumnResult.result.errorMessage) {
```
Comment (abist) [inline extensions/import/src/wizard/pages/summaryPage.ts]: you might want to check for `changeColumnResult` and `changeColumnResult.result` too or wrap this in a `try/catch`

File: extensions/import/src/wizard/pages/summaryPage.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply extensions/import/src/wizard/pages/summaryPage.ts:126]: Not sure what you're saying. You display the errors to the users right? What are they supposed to do then if they see these errors?

File: extensions/import/src/wizard/pages/summaryPage.ts [hunk 4]
```diff
@@ -154,7 +166,7 @@ export class SummaryPage extends ImportPage {
 		this.statusText.updateProperties({
 			value: updateText
 		});
-		return true;
+		return;
```
Comment (Charles-Gagnon) [inline extensions/import/src/wizard/pages/summaryPage.ts:169]: You can just remove this

**Review by Charles-Gagnon — APPROVED** [review]
As far as fixing the immediate issues this seems fine. But as mentioned I would highly suggest going through the process when there's errors and making sure that users are able to have actionable feedback when they occur and that they're able to go back and fix them if so.

**Review by abist — APPROVED** [review]

**Pull request merged**